### PR TITLE
AIP-69: Add leftover glue of all pieces to Edge Provider

### PR DESCRIPTION
**Pull request by jscheffl**
This PR is a split of https://github.com/apache/airflow/pull/41729 and builds on-top of the following pieces as slices of the overall MVP implementation - to make review easier:

- https://github.com/apache/airflow/pull/42046
- https://github.com/apache/airflow/pull/42047
- #42048
- #42049
- #42050 

It only contains the leftovers needed to complete the MVP - basically all "glue" that is not working or fitting to the pieces of the linkes PRs. If you want to see glue only, please take a look to the LAST commit only,

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: docs/apache-airflow/cli-and-env-variables-ref.rst [hunk 1]
```diff
@@ -39,6 +39,7 @@ contributed by the community providers:
 
 * Celery Executor and related CLI commands: :doc:`apache-airflow-providers-celery:cli-ref`
 * Kubernetes Executor and related CLI commands: :doc:`apache-airflow-providers-cncf-kubernetes:cli-ref`
+* Edge Executor and related CLI commands: :doc:`apache-airflow-providers-edge:cli-ref`
```
Comment (kaxil) [inline docs/apache-airflow/cli-and-env-variables-ref.rst]: Let's remove this for now, and do it separately as Airflow docs currently won't have reference to edge provider

Comment (kaxil) [inline docs/apache-airflow/cli-and-env-variables-ref.rst]:
```suggestion
```

Comment (kaxil) [reply docs/apache-airflow/cli-and-env-variables-ref.rst]: Removed it in https://github.com/apache/airflow/pull/42051/commits/f7d5d939e01bf1a264e345c36becff7105e41917 to verify that the test pass

Comment (kaxil) [reply docs/apache-airflow/cli-and-env-variables-ref.rst]: Follow-up created here: https://github.com/apache/airflow/pull/43136 -- details in description

**Pull request merged**